From 791daaa06616ebdc1c558c4c22d06b9738b8a6c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Fri, 4 Dec 2015 18:39:59 +0100
Subject: [PATCH 9/9] Don't abort gst_omx_video_dec_disable() if there's a
 timeout releasing the buffers taken by the egl_render out port

---
 omx/gstomxvideodec.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index baa6faf..e773787 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -2391,8 +2391,11 @@ gst_omx_video_dec_disable (GstOMXVideoDec * self)
     if (gst_omx_port_set_enabled (out_port, FALSE) != OMX_ErrorNone)
       return FALSE;
     if (gst_omx_port_wait_buffers_released (out_port,
-            1 * GST_SECOND) != OMX_ErrorNone)
+            1 * GST_SECOND) != OMX_ErrorNone) {
+#if !(defined (USE_OMX_TARGET_RPI) && defined (HAVE_GST_GL))
       return FALSE;
+#endif
+    }
     if (!gst_omx_video_dec_deallocate_output_buffers (self))
       return FALSE;
     if (gst_omx_port_wait_enabled (out_port, 1 * GST_SECOND) != OMX_ErrorNone)
-- 
2.34.1

